### PR TITLE
use kwargs in get_repository(), get_repos_for_project()

### PR DESCRIPTION
**Pull request by vladak**
This is a follow-up to #3025, making the changes suggested by @ericsaintetienne and adding better tests.

### CODE REVIEW

File: opengrok-tools/src/test/python/test_mirror.py [hunk 1]
```diff
@@ -318,7 +319,7 @@ def mock_get_repo_type(*args):
             git_repo = repos[0]
             assert git_repo.timeout == timeout
             assert git_repo.command == git_cmd_path
-            assert git_repo.env == proxy_dict  # XXX contains
+            assert git_repo.env.items() >= proxy_dict.items()
```
[the remark this answers is not in the document]
Comment (vladak) [reply opengrok-tools/src/test/python/test_mirror.py]: Oh, I did not realize that. I guess this is fine, it's just for the test. Will mention this in the README.

File: opengrok-tools/src/test/python/test_mirror.py [hunk 2]
```diff
@@ -311,11 +313,14 @@ def mock_get_repo_type(*args):
                                           commands=commands,
                                           proxy=proxy_dict,
                                           command_timeout=timeout)
-            print(repos)
             assert len(repos) == 1
+            assert isinstance(repos[0], GitRepository)
+            git_repo = repos[0]
```
[the remark this answers is not in the document]
Comment (vladak) [reply opengrok-tools/src/test/python/test_mirror.py]: To me it makes more sense this way - that `repos[0]` does not have to be a Git repository instance so I prefer to name it so only after it is certain.

File: opengrok-tools/src/test/python/test_mirror.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ktulinger) [reply opengrok-tools/src/test/python/test_mirror.py]: or you can skip the tests when python 3.6 is not available

we already check for python version:
https://github.com/oracle/opengrok/blob/629ad72373aeb10e39c1a163a442140c1542d777/opengrok-tools/src/main/python/opengrok_tools/sync.py#L49-L52

[the remark this answers is not in the document]
Comment (vladak) [reply opengrok-tools/src/test/python/test_mirror.py]: yeah, that's very coarse grind check. I think with the way Python develops we need to cut the line somewhere.

**Pull request merged**